lib/sim.c

```diff
@@ -892,14 +904,18 @@ sim_update_parental_ancestry(sim_t *self, unsigned int parent,
  * Generates parental individuals and inserts them into the parent
  * buffer based on the individuals in the child_buffer. 
  */
+
 static int 
-sim_generate_parents(sim_t *self)
+sim_generate_parents(sim_t *self, int arg_flag)
 {
     int ret = 0;
-    unsigned int j, parent, previous_locus;
+    unsigned int i = 0, j, parent, previous_locus, break_point = 0;
```

Comment:
Do these variables need to be initialised here? It's usually a bit clearer if the variables are initialised next to where they're used. For example, we might initiatialise total_rate below just above the while loop so it's easy to see the initial value of the variable without going to the top of the function.